content/300-guides/300-upgrade-guides/200-upgrading-versions/600-upgrading-to-prisma-5.mdx

```diff
@@ -0,0 +1,430 @@
+---
+title: 'Upgrade to Prisma 5'
+metaTitle: 'Upgrade to Prisma 5'
+metaDescription: 'Guides on how to upgrade to Prisma 5'
+tocDepth: 3
+toc: true
+---
+
+<TopBlock>
+
+Prisma 5.0.0 introduces a number of changes, including the usage of our new JSON Protocol, [which make Prisma faster by default](https://www.prisma.io/blog/prisma-and-serverless-73hbgKnZ6t#improved-startup-performance-by-9x)! A list of these changes can be found [in our release notes](https://github.com/prisma/prisma/releases/tag/5.0.0).
+
+This guide explains how upgrading might affect your application and gives instructions on how to handle breaking changes within Prisma 5.
+
+</TopBlock>
+
+## Breaking changes
+
+This section gives an overview of major breaking changes in Prisma 5.
+
+### Node.js minimum version change
+
+From Prisma version 5.0.0, the minimum version of Node.js supported is 16.13.0. If your project uses an earlier version of Node.js, you will need to upgrade it.
+
+See our [system requirements](/reference/system-requirements) for all minimum version requirements.
+
+<Admonition type="info">
+
+Node.js v16.x is reaching [end-of-life on 11 September 2023](https://nodejs.org/en/blog/announcements/nodejs16-eol) in order to coincide with the end-of-life of OpenSSL 1.1.1. For that reason, we recommend upgrading to the current Node.js LTS, v18.x. Please note that Prisma 5 will be the last major version of Prisma to support Node.js v16.
+
+</Admonition>
+
+### PostgreSQL minimum version change
+
+From Prisma version 5.0.0, the minimum version of PostgreSQL supported is version 9.6. If your project uses an earlier version of PostgreSQL, you will need to update it.
+
+### Removal of array shortcuts
+
+Prisma version 5.0.0 drops support for a number of array shortcuts. These shortcuts were a way to add a single element as a value to an array-based operator instead of an array of one element. In an effort to make our typings more consistent, we now require array values for these operators.
+
+In most cases, the fix will be as simple as wrapping the existing value in an array. Below are some specific examples of needed changes in Prisma 5.
+
+#### <inlinecode>OR</inlinecode> operators
+
+The following code in Prisma 4 and lower:
+
+```js
+prisma.user.findMany({
+  OR: { email: 'foo@example.com' },
+})
+```
+
+Will need to be changed to the following in Prisma 5:
+
+```js highlight=2;edit
+prisma.user.findMany({
+  OR: [{ email: 'foo@example.com' }],
+})
+```
+
+`OR` operators will only accept array values.
+
+#### <inlinecode>in</inlinecode> and <inlinecode>notIn</inlinecode> operators
+
+Similar to `OR`, `in` and `notIn` require array values.
+
+Prisma 4 and lower:
+
+```js
+prisma.user.findMany({
+  id: { in: 123 },
+})
+
+prisma.user.findMany({
+  id: { notIn: 123 },
+})
+```
+
+Prisma 5:
+
+```js highlight=3,9;edit
+prisma.user.findMany({
+  id: {
+    in: [123],
+  },
+})
+
+prisma.user.findMany({
+  id: {
+    notIn: [123],
+  },
+})
+```
```

Comment:
```suggestion
```js highlight=4,12;edit
prisma.user.findMany({
  where: {
    id: {
      in: [123],
    },
  }
})

prisma.user.findMany({
  where: {
    id: {
      notIn: [123],
    },
  }
})
```
```